@react-aria/optimize-locales-plugin does not work on Windows

### Provide a general summary of the issue here

When using `esbuild` and adding the following entry to plugins:

```javascript
optimizeLocales.esbuild({
  locales: ['en-US'],
})
```

I get the following error:

```
X [ERROR] Do not know how to load path: locales-plugin:E:\dev\project\node_modules\@react-aria\optimize-locales-plugin/empty.js

    node_modules/react-aria-components/dist/intlStrings.mjs:1:32:
      1 │ import $bK1z9$arAEmodulejs from "./ar-AE.mjs";
        ╵                                 ~~~~~~~~~~~~~
```

This is due to this line: https://github.com/adobe/react-spectrum/blob/e14088a7edc44422a1135a9412a99cbbc96c1518/packages/dev/optimize-locales-plugin/LocalesPlugin.js#L27

Windows uses backslashes so if you simply concatenate strings you'll get malformed paths. You need to use the node `path` module.

### 🤔 Expected Behavior?

It should not throw errors.

### 😯 Current Behavior

It throws an error because the path is malformed.

### 💁 Possible Solution

Use `path.join` or `path.resolve`.

### 🔦 Context

_No response_

### 🖥️ Steps to Reproduce

You just need to run esbuild with the plugin on Windows.

```
import {build} from 'esbuild';
import optimizeLocales from '@react-aria/optimize-locales-plugin';

build({
  plugins: [
    optimizeLocales.esbuild({
      locales: ['en-US', 'fr-FR']
    })
  ]
});
```

### Version

@react-aria/optimize-locales-plugin v1.1.3

### What browsers are you seeing the problem on?

Other

### If other, please specify.

Node v21.7.1

### What operating system are you using?

Windows 11

### 🧢 Your Company/Team

_No response_

### 🕷 Tracking Issue

_No response_

--- a/packages/dev/optimize-locales-plugin/LocalesPlugin.js
+++ b/packages/dev/optimize-locales-plugin/LocalesPlugin.js
@@ -10,6 +10,7 @@
  * governing permissions and limitations under the License.
  */
 const {createUnplugin} = require('unplugin');
+const path = require('path');
 
 module.exports = createUnplugin(({locales}) => {
   locales = locales.map(l => new Intl.Locale(l));
@@ -24,7 +25,7 @@ module.exports = createUnplugin(({locales}) => {
       if (match) {
         let locale = new Intl.Locale(match[0]);
         if (!locales.some(l => localeMatches(locale, l))) {
-          return __dirname + '/empty.js';
+          return path.join(__dirname, 'empty.js');
         }
       }
 

